### DISCUSSION TITLE
Questions about the aspire dashboard Dockerfile

### DISCUSSION THREAD

**Discussion by WeihanLi** [dotnet/dotnet-docker#5907] [Q&A] [answered]
From the aspire-dashboard Dockerfile

- https://github.com/dotnet/dotnet-docker/blob/08e0e7c3345a75af15ca0c1733d4f07676e0997d/src/aspire-dashboard/8.1/cbl-mariner-distroless/amd64/Dockerfile
- https://github.com/dotnet/dotnet-docker/blob/08e0e7c3345a75af15ca0c1733d4f07676e0997d/src/aspire-dashboard/8.1/cbl-mariner-distroless/arm64v8/Dockerfile

```
ENV \
    # Unset ASPNETCORE_HTTP_PORTS from base image
    ASPNETCORE_HTTP_PORTS= \
    # Aspire Dashboard environment variables
    ASPNETCORE_URLS=http://0.0.0.0:18888 \
    DOTNET_DASHBOARD_OTLP_ENDPOINT_URL=http://0.0.0.0:18889 \
    DOTNET_DASHBOARD_OTLP_HTTP_ENDPOINT_URL=http://0.0.0.0:18890

ENTRYPOINT [ "dotnet", "/app/Aspire.Dashboard.dll" ]
```

I have a question about the `ENV` step, currently we unset the `ASPNETCORE_HTTP_PORTS` environment variable from base image and configure the `ASPNETCORE_URLS` instead, why not configure the `ASPNETCORE_HTTP_PORTS` to `18888` directly?

**Comment by lbussell** [comment, marked as answer]
It is because Aspire Dashboard currently has an issue using ASP.NET Core's "any IP" syntax. You can see more info here: https://github.com/dotnet/dotnet-docker/issues/5190. Should be fixed in Aspire Dashboard 9.0.

**Reply by WeihanLi** [reply]
Thanks very much

**Reply by WeihanLi** [reply]
Seemed it still remains in aspire-dashboard 9.1

**Reply by lbussell** [reply]
Ah, #5190 just tracked changing the URLS syntax from `http://0.0.0.0:18888` to `http://+:18888`

Whereas you're saying it should be `ASPNETCORE_HTTP_PORTS=18888` instead.

https://github.com/dotnet/dotnet-docker/blob/8bf45708a518d853ffa7308d7f8663e4bd6e70e8/src/aspire-dashboard/9.1/azurelinux-distroless/amd64/Dockerfile#L27-L33

/cc @joperezr if you want to fix this in nightly.
